x-pack/plugins/beats_management/public/bootstrap.tsx

```diff
@@ -18,10 +18,13 @@ import { BeatsManagementConfigType } from '../common';
 import { MANAGEMENT_SECTION } from '../common/constants';
 
 async function startApp(libs: FrontendLibs, core: CoreSetup<StartDeps>) {
-  const [startServices] = await Promise.all([
-    core.getStartServices(),
-    libs.framework.waitUntilFrameworkReady(),
-  ]);
+  const startServices = await core.getStartServices();
+
+  if (startServices[0].http.anonymousPaths.isAnonymous(window.location.pathname)) {
+    return;
+  }
+  // Can't run until the `start` lifecycle, so we wait for start services to resolve above before calling this.
+  await libs.framework.waitUntilFrameworkReady();
```

Comment:
note: this isn't strictly related to the resolved issues, but I noticed that Beats was making requests which required authentication on the login page (and other unauthenticated pages). This change prevents Beats from completing its initialization if we are on one such page. Moving from an unauthenticated page to an authenticated page requires a full page reload, so there is no risk to the actual stack management application.